bip-templatehash.md

```diff
@@ -0,0 +1,171 @@
+following:
+- *hash_type*: this is the sighash type identifier. Only a single hash type is supported by `OP_TEMPLATEHASH`, so there
+  is no need to commit to such an identifier.
+- *spend_type*: this value is defined by BIP341 as *2\*ext_flag + annex_present*. Since no extension is appended to the
+  signature message, *ext_flag* is set to 0. Therefore we commit directly to *annex_present*.
+- *sha_prevouts* / *sha_scriptpubkeys*: committing to these fields as is would introduce a hash cycle when the hash is
+  committed in the output itself. Committing to all other prevouts or scriptpubkeys would introduce hashing a quantity
+  of data quadratic in the number of inputs. It would also prevent spending two coins encumbered by a template hash
+  check in the same transaction. Finally, the flexibility of not committing to the specific coins spent is also
+  desirable to recover from mistakes[^no-commit-other-coins].
+- *sha_amounts*: the BIP341 rationale for committing to the amounts of all spent coins is to be able to prove to an offline
+  signer the fees of a transaction. Although `OP_TEMPLATEHASH` can be used as a building block for rebindable
+  signatures, the utility of committing to spent amounts but not spent scriptpubkeys [is
+  limited][greg-attack-input-ownership]. Still for rebindable signatures, committing to spent amounts can be justified
+  as defense-in-depth against implementation mistakes[^commit-spent-amounts]. However, the lack of flexibility this
+  introduces also makes it harder to recover from a mistake in committing to the next
+  transaction[^no-commit-other-coins]. Furthermore, committing to all spent amounts also makes overcommitting funds to
+  such a script result in the output being forever unspendable instead of the excess just going to fees at spend time.
+
+The design of `OP_TEMPLATEHASH` was inspired by the design of [BIP119][ctv-bip] `OP_CHECKTEMPLATEVERIFY` but differs in
+several important ways.
+
+First of all, `OP_TEMPLATEHASH` is only defined for Tapscript, as modifying legacy Script comes with an unnecessarily
+increased risk surface. In addition, sticking to Tapscript allows leveraging more powerful upgrade hooks (`OP_SUCCESS`s
+instead of `OP_NOP`s) which make it possible to push the template hash on the stack instead of being constrained to
+strict assertions with no stack modification. Pushing the template hash on the stack substantially improves the
+efficiency of using `OP_TEMPLATEHASH` as a building block for rebindable signatures.
+
+Unlike `OP_TEMPLATEHASH`, `OP_CHECKTEMPLATEVERIFY` also commits to the scriptSig of all inputs of the spending
+transaction. `OP_CHECKTEMPLATEVERIFY` gives txid stability when the committed spending transaction has a single input,
+and when the scriptSig of this single input has been committed by the hash.
+Taproot scriptSigs must be empty and therefore under the single input case `OP_TEMPLATEHASH` has no requirement
+to commit to scriptSigs to achieve txid stability.
```

Comment:
everytime I read through this always my biggest question is "why not commit to all script sigs being empty" or "what are the potential downsides to having multi input malleability"

would be nice to have an explanation in this to explain why this trade off is okay